include/nbl/builtin/hlsl/math/complex.hlsl

```diff
@@ -0,0 +1,110 @@
+
+// Copyright (C) 2018-2022 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_HLSL_MATH_COMPLEX_INCLUDED_
+#define _NBL_BUILTIN_HLSL_MATH_COMPLEX_INCLUDED_
+
+#include <nbl/builtin/hlsl/math/constants.hlsl>
+#include <nbl/builtin/hlsl/math/functions.hlsl>
+
+
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace math
+{
+namespace complex
+{
+
+
+#define complex16_t uint
+
+#define complex float2
+#define cfloat2 float2x2
+#define cfloat3 mat2x3
+#define cfloat4 mat2x4
```

Comment:
we can use `typedef` for these now!

also repalce `mat2x3` with `float2x3`, etc.

Reply:
actually make a `struct` for each of them

Reply:
for `complex_t<float>` and similar, `real` is the old `x` component` and `imaginary` is `y`